extensions/sld/src/BusbarSectionPositionAdder.cpp

```diff
@@ -0,0 +1,56 @@
+/**
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#include <powsybl/iidm/extensions/sld/BusbarSectionPositionAdder.hpp>
+
+#include <powsybl/PowsyblException.hpp>
+#include <powsybl/iidm/BusbarSection.hpp>
+#include <powsybl/iidm/extensions/sld/BusbarSectionPosition.hpp>
+#include <powsybl/stdcxx/format.hpp>
+
+namespace powsybl {
+
+namespace iidm {
+
+namespace extensions {
+
+namespace sld {
+
+BusbarSectionPositionAdder::BusbarSectionPositionAdder(Extendable& extendable) :
+    ExtensionAdder(extendable) {
+}
+
+std::unique_ptr<Extension> BusbarSectionPositionAdder::createExtension(Extendable& extendable) {
+    if (!m_busbarIndex.is_initialized()) {
+        throw PowsyblException("Undefined value for busbar index");
+    }
+    if (!m_sectionIndex) {
+        throw PowsyblException("Undefined value for section index");
+    }
```

Comment:
Choose the way you want to check indexes are set, but use the same syntax wherever. It seems we are not consistent in the API. Could you also create an issue to fix these inconsistencies?